dataactbroker/app.py

```diff
@@ -58,24 +56,14 @@ def createApp():
         app.session_interface = DynamoInterface()
         # Set up bcrypt
         bcrypt = Bcrypt(app)
-        def clearInterfaces(response):
-            try:
-                interfaces =BaseInterface.interfaces
-                interfaces.jobDb.close()
-                interfaces.validationDb.close()
-                interfaces.errorDb.close()
-                interfaces.userDb.close()
-                BaseInterface.interfaces = None
-            except Exception as e:
-                print("Could not close connections")
-                print(str(type(e)) + ": " + str(e))
-                pass
-            return response
-        app.after_request(clearInterfaces)
-
-        def createInterfaces():
-            BaseInterface.interfaces = InterfaceHolder()
-        app.before_request(createInterfaces)
+
+        @app.teardown_appcontext
+        def teardown_appcontext(exception):
+            GlobalDB.close()
+
+        @app.before_request
+        def before_request():
+            GlobalDB.db()
```

Comment:
I would agree that I like the explicitness, it makes it very clear what's going on